### DISCUSSION TITLE
How can static assets be cached?

### DISCUSSION THREAD

**Discussion by dbzx10299** [nuxt/nuxt#27169] [Questions] [answered]
How can I cache a static asset such as a font or a stylesheet? I know there are route rules, but I am a bit confused how to go about doing this since when the build step is ran the output has a different hash.

**Comment by TheAlexLichter** [comment]
Everything in `_nuxt`, which are all static assets, is cached by default already through the route rules ☺️

**Reply by dbzx10299** [reply]
So would we be able to create a route rule something like this then?

```js
export default defineNuxtConfig({
  routeRules: {
    '/_nuxt': {
        headers: {
          'cache-control': ''
        }
    }
  }
})
```

It‘s nice nuxt does this by default but I would like to know a bit better how this could be done manually. 

The above example isn‘t the best I guess but I‘m wondering hypothetically if we had to cache an asset manually in the public folder how that might look.

**Reply by TheAlexLichter** [reply, marked as answer]
More or less like:

```ts
export default defineNuxtConfig({
  routeRules: {
    '/_nuxt/**': {
        headers: {
          'cache-control': 'whatever-you-want-here'
        }
    }
  }
})
```

Related: https://github.com/nuxt/nuxt/issues/15465

**Reply by dbzx10299** [reply]
Got it, thanks!

**Reply by dbzx10299** [reply]
One more question regarding this automatic caching that nuxt performs for assets, does nuxt automatically set the cache headers for a netlify runtime?

**Reply by TheAlexLichter** [reply]
It should. You can check it in your dev tools.

<img width="523" alt="image" src="https://github.com/nuxt/nuxt/assets/640208/41e3f959-fed6-45c1-8519-f9950ae8ab34">

**Reply by dbzx10299** [reply]
Netlify has some of their own cache headers so I guess those would be set as you specified earlier with route rules

https://docs.netlify.com/platform/caching/#supported-cache-control-headers
